File: backend/app/schemas/habitacion.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict


class TipoHabitacionNested(BaseModel):
    id: int
    nombre: str
    precio_dia: float
    capacidad: int

    model_config = ConfigDict(from_attributes=True)


class HabitacionBase(BaseModel):
    piso: str
    numero: str
    tipo_habitacion_id: int
    ocupado: bool = False


class HabitacionCreate(HabitacionBase):
    pass


class HabitacionUpdate(BaseModel):
    piso: Optional[str] = None
    numero: Optional[str] = None
    tipo_habitacion_id: Optional[int] = None
    ocupado: Optional[bool] = None


class HabitacionResponse(HabitacionBase):
    id: int
    tipo_habitacion: Optional[TipoHabitacionNested] = None
    created_at: datetime
    updated_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)


class HabitacionDisponibilidad(BaseModel):
    id: int
    piso: str
    numero: str
    tipo_habitacion: TipoHabitacionNested

    model_config = ConfigDict(from_attributes=True)


class HabitacionDisponibilidadResponse(BaseModel):
    habitacion: HabitacionDisponibilidad
    disponible: bool

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/habitacion.ts
import { TipoHabitacionNested } from './tipoHabitacion';

export interface Habitacion {
  id: number;
  piso: string;
  numero: string;
  tipo_habitacion_id: number;
  ocupado: boolean;
  tipo_habitacion?: TipoHabitacionNested;
  created_at: string;
  updated_at?: string;
}

export interface HabitacionCreate {
  piso: string;
  numero: string;
  tipo_habitacion_id: number;
  ocupado?: boolean;
}

export interface HabitacionUpdate {
  piso?: string;
  numero?: string;
  tipo_habitacion_id?: number;
  ocupado?: boolean;
}

export interface HabitacionDisponibilidad {
  id: number;
  piso: string;
  numero: string;
  tipo_habitacion: TipoHabitacionNested;
}

export interface HabitacionDisponibilidadResponse {
  habitacion: HabitacionDisponibilidad;
  disponible: boolean;
}

export interface OcupacionStats {
  total_habitaciones: number;
  habitaciones_ocupadas: number;
  habitaciones_disponibles: number;
  reservas_activas: number;
  porcentaje_ocupacion: number;
}
